torch/autograd/gradcheck.py

```diff
@@ -15,17 +15,26 @@ def zero_gradients(x):
         for elem in x:
             zero_gradients(elem)
 
-
-def make_jacobian(input, num_out):
+# Returns a tensor or iterable of tensors each with shape
+# (input.nelement(), num_out) if input is of non-complex dtype
+# and of shape (input.nelement(), num_out, 2) for tensors with complex dtype when extra_dim_if_complex
+# is set to True. The extra dimension helps store both wirtinger and conjugate wirtinger derivative
+# required for gradcheck for complex functions.
+# For R -> C, wirtinger and conjugate wirtinger derivative are equal.
+def make_jacobian(input, num_out, extra_dim_if_complex=False):
     if isinstance(input, torch.Tensor):
         if not input.is_floating_point() and not input.is_complex():
             return None
         if not input.requires_grad:
             return None
-        return torch.zeros(input.nelement(), num_out, dtype=input.dtype)
+        shape = (input.nelement(), num_out)
+        # two dimensions to store dout_dx, dout_dy for an input x + yj
```

Comment:
The comment above mentions that these contain wirtinger and conjugate wirtinger. These are not the same as derivatives wrt to real and imaginary parts respectively right?

Reply:
no forgot to update the comment after making some changes! updated